### PR TITLE
feat(playwright): add big sur beta support

### LINKED ISSUE #2789: [Feature] macOS Big Sur support
Let us know what functionality you'd like to see in Playwright and what is your use case.
With the looming release of macOS Big Sur, playwright should support it.

Do you think others might benefit from this as well?
There could be lots of people upgrading when the time comes.

### PR DESCRIPTION
**Pull request by p0x6**
Adding mac10.16 to be supported. This is the macOS Big Sur beta version.

Begins to fix: #2789

### CODE REVIEW

File: src/install/browserFetcher.ts [hunk 1]
```diff
@@ -84,6 +86,7 @@ function getDownloadUrl(browserName: BrowserName, platform: BrowserPlatform): st
       ['mac10.13', undefined],
       ['mac10.14', '%s/builds/webkit/%s/minibrowser-mac-10.14.zip'],
       ['mac10.15', '%s/builds/webkit/%s/minibrowser-mac-10.15.zip'],
+      ['mac10.16', '%s/builds/webkit/%s/minibrowser-mac-10.15.zip'],
```
Comment (yury-s) [inline src/install/browserFetcher.ts:89]: I believe this requires a dedicated build similar to 10.14. We shouldn't provide a link for now unless it's working.

File: src/install/browserFetcher.ts [hunk 2]
```diff
@@ -73,6 +74,7 @@ function getDownloadUrl(browserName: BrowserName, platform: BrowserPlatform): st
       ['mac10.13', '%s/builds/firefox/%s/firefox-mac.zip'],
       ['mac10.14', '%s/builds/firefox/%s/firefox-mac.zip'],
       ['mac10.15', '%s/builds/firefox/%s/firefox-mac.zip'],
+      ['mac10.16', '%s/builds/firefox/%s/firefox-mac.zip'],
```
Comment (yury-s) [inline src/install/browserFetcher.ts:77]: I don't have a Big Sur machine at hand. Does our current binary just work on 10.16?

**Review by yury-s — COMMENTED** [review]
Thanks for your contribution!

File: src/install/browserFetcher.ts (hunk 2 above)
Comment (p0x6) [reply src/install/browserFetcher.ts:77]: it does. just upgraded my dev machine to big sur yesterday.

which is why i opened this pr.

File: src/install/browserFetcher.ts (hunk 1 above)
Comment (p0x6) [reply src/install/browserFetcher.ts:89]: tested on my big sur dev machine :)

Comment (p0x6) [reply src/install/browserFetcher.ts:89]: it could change when the actual big sur release is out.

File: utils/check_availability.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ const browserFetcher = require('../lib/install/browserFetcher.js');
 const https = require('https');
 const SUPPORTER_PLATFORMS = ['linux', 'mac', 'win32', 'win64'];
 
-const fetcherOptions = SUPPORTER_PLATFORMS.map(platform => platform === 'mac' ? 'mac10.15' : platform);
+const fetcherOptions = SUPPORTER_PLATFORMS.map(platform => platform === 'mac' ? 'mac10.16' : platform);
```
Comment (aslushnikov) [inline utils/check_availability.js:23]: I believe this is not really needed?

**Comment by aslushnikov** [comment]
@p0x6 Thank you for the PR!

Just to re-iterate: do I understand correctly that all our Chromium, Firefox and WebKit actually work for you on Big Sur?

File: utils/check_availability.js (hunk 3 above)
Comment (p0x6) [reply utils/check_availability.js:23]: will remove this

**Comment by p0x6** [comment]
> @p0x6 Thank you for the PR!
> 
> Just to re-iterate: do I understand correctly that all our Chromium, Firefox and WebKit actually work for you on Big Sur?

You are correct. I am running playwright tests on Big Sur beta (mac10.16) with my changes. Using the current binaries.

I am not sure if it will be the case when Big Sur is officially released as mac11.0, but as of now the beta works.

**Comment by aslushnikov** [comment]
@p0x6 we discussed this offline with the team. It was decided that we can't add BigSur without bots support, and currently there are no BigSur machines in either of the CI providers we use.

So I'll have to refrain from merging this; we'll probably get back to this in fall once BigSur is officially released.

**Comment by mkleehammer** [comment]
Is the reason just support?  There are a lot of devs on the beta.  Couldn't we merge this with a warning on the console that Big Sur is not officially supported?

**Comment by vsgopinath** [comment]
Hey Playwright team,
      I am just using the mac11.0 Big Sur Beta. Is it possible to have the support for Mac 11.0 too?
Thanks in advance

**Pull request closed without merging**